ebos/eclgenericproblem_impl.hh

```diff
@@ -367,19 +367,19 @@ updateNum(const std::string& name, std::vector<T>& numbers, std::size_t num_regi
     if (!eclState_.fieldProps().has_int(name))
         return;
 
-    const auto& numData = eclState_.fieldProps().get_int(name);
-
     unsigned numElems = gridView_.size(/*codim=*/0);
-    numbers.resize(numElems);
-    for (unsigned elemIdx = 0; elemIdx < numElems; ++elemIdx) {
-        if (numData[this->getLookUpData(elemIdx)] > (int)num_regions) {
-            throw std::runtime_error("Values larger than maximum number of regions " + std::to_string(num_regions) + " provided in " + name);
-        } else if (numData[this->getLookUpData(elemIdx)] > 0) {
-            numbers[elemIdx] = static_cast<T>(numData[this->getLookUpData(elemIdx)]) - 1;
-        } else {
+    std::function<void(T)> valueCheck = [num_regions,name](T number) {
+        if ( (number+1) > (int)num_regions) {
+            throw std::runtime_error("Values larger than maximum number of regions "
+                                     + std::to_string(num_regions) + " provided in " + name);
+        }
+        if ( (number+1) <= 0) {
             throw std::runtime_error("zero or negative values provided for region array: " + name);
         }
-    }
+    };
+
+    numbers = this->lookUpData_.assignFieldPropsIntOnLeaf(eclState_.fieldProps(), name, numElems, 1 /*defaultValue*/,
```

Comment:
I think 1 might not be correct here ( with the current version of the other PR), as the default value is not translated in the companion PR.  Of course all default values will be overwritten anyway...

Maybe we should do the translation in the other PR? If we also do the checks with the untranslated values above then this is consistent.

Reply:
It seems I could get the default values with an unordered map (https://github.com/OPM/opm-common/blob/aaa8af923ceeaaff60e788c224b86c77ea685498/opm/input/eclipse/EclipseState/Grid/FieldProps.hpp#L241). I agree that it must be better to include that in the LookUpData code. Thanks!

Reply:
I probably was the one suggesting that there should be a default value. Not sure what triggered me to do that. :see_no_evil: 
These field properties should always have a value for all the cells (here on the coarse grid). So it seems we do not need to pass a default value to that function as everything gets overwritten anyway.

The code you are pointing to, will make sure that if certain field properties are omitted then we can still get the array for the field property with the correct default value used for all cells.